http/middleware.go

```diff
@@ -62,42 +64,35 @@ func processRequest(tx types.Transaction, req *http.Request) (*types.Interruptio
 		// body inspection, otherwise we just let the request follow its
 		// regular flow.
 		if req.Body != nil && req.Body != http.NoBody {
-			_, err := io.Copy(tx.RequestBodyWriter(), req.Body)
+			b, err := io.ReadAll(req.Body)
 			if err != nil {
-				return tx.Interruption(), err
+				return nil, fmt.Errorf("failed to read request body: %s", err.Error())
 			}
 			_ = req.Body.Close()
 
-			reader, err := tx.RequestBodyReader()
+			in, limit, err := tx.ProcessRequestBodySize(len(b))
 			if err != nil {
-				return tx.Interruption(), err
+				return nil, fmt.Errorf("failed to process request body size: %s", err.Error())
 			}
-			// req.Body is transparently reinizialied with a new io.ReadCloser.
-			// The http handler will be able to read it.
-			// Prior to Go 1.19 NopCloser does not implement WriterTo if the reader implements it.
-			// - https://github.com/golang/go/issues/51566
-			// - https://tip.golang.org/doc/go1.19#minor_library_changes
-			// This avoid errors like "failed to process request: malformed chunked encoding" when
-			// using io.Copy.
-			// In Go 1.19 we just do `req.Body = io.NopCloser(reader)`
-			if rwt, ok := reader.(io.WriterTo); ok {
-				req.Body = struct {
-					io.Reader
-					io.WriterTo
-					io.Closer
-				}{reader, rwt, nopCloser{}}
-			} else {
-				req.Body = struct {
-					io.Reader
-					io.Closer
-				}{reader, nopCloser{}}
+			if in != nil {
+				return in, nil
 			}
-		}
 
-		return tx.ProcessRequestBody()
+			_, err = tx.RequestBodyWriter().Write(b[:limit])
```

Comment:
This is important, we only copy to body writer the max number of bytes aimed to analysis instead of the whole thing.